src/components/CollaboratorModal/components/MainSubmodal.tsx

```diff
@@ -0,0 +1,251 @@
+              </Box>
+            </GridItem>
+            <GridItem colSpan={2}>
+              <Box display="flex" alignItems="center" h="100%">
+                <Text textTransform="capitalize">
+                  {_.capitalize(collaborator.role)}
+                </Text>
+              </Box>
+            </GridItem>
+            <GridItem colSpan={1}>
+              <Box display="flex" alignItems="center" h="100%">
+                <IconButton
+                  aria-label="Delete collaborator button"
+                  variant="clear"
+                  colorScheme="danger"
+                  onClick={() => onDelete(collaborator)}
+                  id={`delete-${collaborator.id}`}
+                  icon={<BiTrash color="icon.danger" />}
+                  isDisabled={isDisabled || isError}
+                />
+              </Box>
+            </GridItem>
+          </Grid>
+          <Divider />
+        </>
+      )) ?? (
+        <Stack>
+          {Array(3)
+            .fill(null)
+            .map(() => (
+              <Skeleton height="2rem" />
+            ))}
+        </Stack>
+      )}
+    </Box>
+  )
+}
+
+const extractErrorMessage = (props: MiddlewareError | undefined): string => {
+  if (!props || props?.code === 500) return DEFAULT_RETRY_MSG
+
+  return props.message
+}
+
+interface MainSubmodalProps extends Omit<ModalProps, "children"> {
+  onDelete: (user: Collaborator) => void
+}
+
+export const MainSubmodal = ({
+  onDelete,
+  ...props
+}: MainSubmodalProps): JSX.Element => {
+  const { siteName } = useParams<{ siteName: string }>()
+  const [showAck, setShowAck] = useState(false)
+  const successToast = useSuccessToast()
+  const {
+    mutateAsync: addCollaborator,
+    error: addCollaboratorError,
+    isSuccess: addCollaboratorSuccess,
+    isError: isAddCollaboratorError,
+    isLoading: isAddCollaboratorLoading,
+    reset,
+  } = CollaboratorHooks.useAddCollaboratorHook(siteName)
+  const { data: role } = CollaboratorHooks.useGetCollaboratorRoleHook(siteName)
+
+  const errorMessage = extractErrorMessage(
+    addCollaboratorError?.response?.data.error
+  )
+
+  const collaboratorFormMethods = useForm({
+    mode: "onTouched",
+    defaultValues: {
+      newCollaboratorEmail: "",
+      isAcknowledged: false,
+    },
+  })
+
+  // Show acknowledgement modal if user does not have a trusted email
+  useEffect(() => {
+    // No error message implies no error
+    if (!addCollaboratorError?.response?.data?.error) return
+
+    const { message: errMessage } = addCollaboratorError?.response?.data?.error
+
+    if (errMessage === ACK_REQUIRED_ERROR_MESSAGE) {
+      setShowAck(true)
+    }
+  }, [addCollaboratorError])
+
+  useEffect(() => {
+    if (addCollaboratorSuccess) {
+      successToast({ description: "Collaborator added successfully" })
+      collaboratorFormMethods.reset()
+    }
+  }, [addCollaboratorSuccess, collaboratorFormMethods, successToast])
+
+  const isDisabled = role !== "ADMIN"
```

Comment:
I think we should also disable the button if the input field is empty or invalid.

Reply:
good point - added on the button; this is also used for the form control, which will disable the input as well.